### PR TITLE
Try merging back java support

### PR DESCRIPTION
**Pull request by cvbge**
Hi, I noticed you added java support, would you be interested in merging it back?

### CODE REVIEW

**Comment by cvbge** [comment]
@MarcusGSattler

**Comment by MarcusGSattler** [comment]
Made some quick merge conflict fixes, but these changes are untested.

File: src/extension.ts [hunk 1]
```diff
@@ -61,9 +61,20 @@ export function escapeString(s: string, languageId: string, eol: string)
...
+        } else if (languageId === "java") {
+        const escaped = s
+            .replace(/\\/g, "\\\\") // backslashes first
+            .replace(/"/g, "\\\"")
+            .replace(/'/g, "\\'")
+            .replace(/\n/g, "\\n")
+            .replace(/\r/g, "\\r")
+            .replace(/\t/g, "\\t")
+            .replace(/\f/g, "\\f");
+        return escaped.split(/\r?\n/).map((s, idx, arr) => `"${s}${idx !== arr.length - 1 ? '\\n' : ''}"`).join(eol);
```
Comment (cvbge) [inline src/extension.ts:74]: I guess Java does not have "raw" string literals?

Are all those replacements needed? I mean, isn't it enough to escape only backslashes and quote marks (") ?

About special characters like \n or \t, I guess it's questionable what's the better way. Maybe it should be optional. But currently other languages don't escape them...

File: package.json [hunk 2]
```diff
@@ -1,12 +1,12 @@
 {
-  "name": "escape-string",
-  "displayName": "Escape String",
-  "description": "Escape string when pasting",
-  "version": "1.1.0",
-  "publisher": "cvbge",
+  "name": "escape-string-for-java",
```
Comment (cvbge) [inline package.json:2]: I can't merge it like that...

**Comment by cvbge** [comment]
Thanks Marcus,

I think for this merge to happen I'd need following:
- we'd need to agree what to do about special characters like \f, right now my opinion is that it should be consistent with other languages
- only minimal changes (e.g. code to escape string + registration in packages.json + potentially the new keybinding, although ideally it'd be a separate commit)
- clean history

Let me know what you think

**Pull request closed without merging**